src/components/Hosts/HostPopover.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { useRef, useEffect } from "react";
+import styled from "@emotion/styled";
+import Popover from "@leafygreen-ui/popover";
+import { Button } from "components/Button";
+
+interface Props {
+  titleText: string;
+  loading: boolean;
+  onClick: () => void;
+  active: boolean;
+  setActive: (isActive: boolean) => void;
+  "data-cy"?: string;
+}
+
+export const HostPopover: React.FC<Props> = ({
+  titleText,
+  loading,
+  onClick,
+  active,
+  setActive,
+  "data-cy": dataCy,
+}) => {
+  const popoverRef = useRef(null);
+
+  // Handle onClickOutside
+  useEffect(() => {
+    if (!active) {
+      return;
+    }
+    const onClickOutside = (event: MouseEvent) => {
+      if (
+        popoverRef.current &&
+        !popoverRef.current.contains(event.target as Node)
+      ) {
+        setActive(false);
+      }
+    };
+
+    document.addEventListener("mousedown", onClickOutside);
+    return () => {
+      document.removeEventListener("mousedown", onClickOutside);
+    };
+  }, [popoverRef, active]); // eslint-disable-line react-hooks/exhaustive-deps
+
+  return (
+    <StyledPopover
+      align="bottom"
+      justify="middle"
+      active={active}
+      data-cy={dataCy}
+    >
+      <PopoverContainer ref={popoverRef}>
+        {titleText}
+
+        <ButtonContainer>
+          <ButtonSpacer>
+            <Button size="xsmall" disabled={loading}>
+              No
+            </Button>
+          </ButtonSpacer>
+          <ButtonSpacer>
+            <Button
+              variant="primary"
+              size="xsmall"
+              disabled={loading}
+              onClick={onClick}
+            >
+              Yes
+            </Button>
+          </ButtonSpacer>
+        </ButtonContainer>
+      </PopoverContainer>
+    </StyledPopover>
+  );
+};
+
+// @ts-expect-error
+const StyledPopover = styled(Popover)`
+  width: 300px;
+`;
+
+const PopoverContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  background-color: white;
+  padding: 16px;
+  box-shadow: 0 3px 6px -4px rgb(0 0 0 / 12%), 0 6px 16px 0 rgb(0 0 0 / 8%),
+    0 9px 28px 8px rgb(0 0 0 / 5%);
```

Comment:
Sorry, I took a little creative freedom because the default `Popover` is pretty hard to see (no border, no shading):

<img width="1668" alt="Screen Shot 2021-11-18 at 8 31 54 PM" src="https://user-images.githubusercontent.com/47064971/142542823-3deade5c-702f-47dd-a9c6-27b862787a0c.png">

I can make this line neater by using the official MongoDB colors from `uiColors.ts`.

Reply:
haha all good. Might also be a good idea to bring it up with leafygreen maybe they have some suggestions in mind.